ldp/alg/rollout.py

```diff
@@ -42,8 +42,21 @@ def reraise_exc_as(reraise: type[CaughtError], enabled: bool) -> Iterator[None]:
         yield
     except Exception as e:
         if enabled:
-            logger.exception(f"Caught {reraise.exc_type} exception.")
-            raise reraise(e) from e
+            # Add more detailed error information
```

Comment:
I like it, and I like how you set it up with duck typing to support multiple possible HTTP error types. The one tough part of duck typed code is making changes are a bit harder, so it would be good to unit test this.

Do you mind

1. Moving this to a free function in `ldp/utils.py`
2. Making a brief unit test of it, with the particular error type we support

Also, feel free to ignore this comment

Reply:
Valid comments, thanks – updated